src/main/java/org/zalando/nakadi/domain/ResourceImpl.java

```diff
@@ -0,0 +1,55 @@
+package org.zalando.nakadi.domain;
+
+import org.zalando.nakadi.plugin.api.authz.AuthorizationAttribute;
+import org.zalando.nakadi.plugin.api.authz.AuthorizationService;
+import org.zalando.nakadi.plugin.api.authz.Resource;
+
+import java.util.List;
+import java.util.Optional;
+
+public class ResourceImpl<T> implements Resource {
+
+    public static final String ALL_DATA_ACCESS_RESOURCE = "all-data-access";
+    public static final String ADMIN_RESOURCE = "nakadi";
+    public static final String EVENT_TYPE_RESOURCE = "event-type";
+    public static final String SUBSCRIPTION_RESOURCE = "subscription";
+
+    private final T resource;
```

Comment:
How this field is used? I could not find usages in Nakadi, so probably it is used in a plugin. Can you please share the link to appropriate code?

Reply:
@v-stepanov it would be used to filter resources according to some criteria implemented in a plugin: https://github.com/zalando-nakadi/nakadi-plugin-api/pull/4